recursion/compiler/src/util.rs

```diff
@@ -0,0 +1,9 @@
+use p3_field::PrimeField32;
+
+pub fn canonical_i32_to_field<F: PrimeField32>(x: i32) -> u32 {
+    if x < 0 {
```

Comment:
It seems like this assumes `-P < x < P` which I think seems fine based on how it's currently used, but maybe good to add an assert in case it gets used for other things later

Reply:
added assert